app/utility/base_world.py

```diff
@@ -109,6 +112,26 @@ async def walk_file_path(path, target):
                 return os.path.join(root, '%s.xored' % target)
         return None
 
+    @staticmethod
+    def check_requirement(params):
+        def parse_version(version_string, pattern=r'([0-9]+(?:\.[0-9]+)+)'):
+            groups = re.search(pattern, version_string)
+            if groups:
+                return groups[1]
+            return '0.0.0'
+
+        try:
+            if 'module' in params:
```

Comment:
ugly function warning ;)

Reply:
hopefully the villagers have not gathered their pitchforks yet, I just pushed a refactor to compartmentalize things and split out functionality